File: Short Long Short/short-long-short.py
'''
OBJECTIVE: given two strings, return the longer string
sandwiched between two copies of the shorter string. If
either string is empty, return two copies of the none-
empty string. If both are empty return ""
'''

def solution(a, b):
    strings = [a, b]
    strings.sort(key=len)
    if a and b:
        return strings[0] + strings[1] + strings[0]
    else:
        return 2 * strings[1]

File: Short Long Short/short-long-short.hs
-- Kata 50654ddff44f800200000007: Short Long Short
-- By:
-- OBJECTIVE: given two strings, return the longer string
-- sandwiched between two copies of the shorter string.

module ShortLongShort where

import Data.List (sortOn)
sortByLen = sortOn length
shortLongShort :: String -> String -> String
shortLongShort a b  = (sortByLen [a, b] !! 0) ++ (sortByLen [a, b] !! 1) ++ (sortByLen [a, b] !! 0)